net/utils/net_ipchksum.c

```diff
@@ -93,13 +94,11 @@ uint16_t ipv4_upperlayer_chksum(FAR struct net_driver_s *dev, uint8_t proto)
 
   /* Sum IP payload data. */
 
-#ifdef CONFIG_MM_IOB
   if (dev->d_iob != NULL)
```

Comment:
should we remove the check? since IPBUF also access d_iob too.

Reply:
Done